src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,156 @@
+        if (member == null || msg.getAuthor().isBot() || msg.getAuthor().isSystem() ||
+            member.hasPermission(Permission.MANAGE_CHANNEL)) {
+            return;
+        }
+        if (containsScam(msg)) {
+            final var guild = msg.getGuild();
+            final var embed = getLoggingEmbed(msg, ", by editing an old message");
+            msg.delete().reason("Scam link").queue($ -> {
+                executeInLoggingChannel(channel -> channel.sendMessageEmbeds(embed).queue());
+                mute(guild, member);
+            });
+        }
+    }
+
+    private static void mute(final Guild guild, final Member member) {
+        final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+        // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+        final var mutedRole = guild.getRoleById(mutedRoleID);
+        if (mutedRole == null) {
+            MMDBot.LOGGER.error(MMDMarkers.MUTING, "Unable to find muted role {}", mutedRoleID);
+            return;
+        }
+        guild.addRoleToMember(member, mutedRole).queue();
+    }
+
+    private static MessageEmbed getLoggingEmbed(final Message message, final String extraDescription) {
+        final var member = message.getMember();
+        return new EmbedBuilder().setTitle("Scam link detected!")
+            .setDescription(String.format("User %s sent a scam link in %s%s. Their message was deleted, and they were muted.", member.getUser().getAsTag(),
+                message.getTextChannel().getAsMention(), extraDescription))
+            .addField("Message Content", MarkdownUtil.codeblock(message.getContentRaw()), false)
+            .setColor(Color.RED)
+            .setTimestamp(Instant.now())
+            .setFooter("User ID: " + member.getIdLong())
+            .setThumbnail(member.getEffectiveAvatarUrl()).build();
+    }
+
+    private static void executeInLoggingChannel(Consumer<TextChannel> channel) {
+        Utils.getChannelIfPresent(MMDBot.getConfig().getChannel("events.requests_deletion"), channel);
+    }
+
+    public static boolean containsScam(final Message message) {
+        final String msgContent = message.getContentRaw().toLowerCase(Locale.ROOT);
+        synchronized (SCAM_LINKS) {
+            for (final var link : SCAM_LINKS) {
+                if (msgContent.contains(link)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static void setupScamLinks() {
+        SCAM_LINKS.clear();
+        MMDBot.LOGGER.debug("Setting up scam links! Receiving data from {}.", SCAM_LINKS_DATA_URL);
+        try (var is = new URL(SCAM_LINKS_DATA_URL).openStream()) {
+            final String result = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+            SCAM_LINKS.addAll(StreamSupport.stream(GSON.fromJson(result, JsonArray.class).spliterator(), false)
+                .map(JsonElement::getAsString).filter(s -> !s.contains("discordapp.co")).toList());
```

Comment:
I feel like matching against domains containing the string "discordapp.co" may potentially exclude some scam domains. Do we actually need this check here?

Reply:
Leaving that domain would mean that it would flag emoji links as scams. I guess I could do a manual check for the emoji link and whitelist it but that may be inefficient